src/Microsoft.OpenApi.OData.Reader/Operation/NavigationPropertyDeleteOperationHandler.cs

```diff
@@ -19,14 +21,23 @@ internal class NavigationPropertyDeleteOperationHandler : NavigationPropertyOper
     {
         /// <inheritdoc/>
         public override OperationType OperationType => OperationType.Delete;
+        private DeleteRestrictionsType _deleteRestriction;
+
+        /// <inheritdoc/>
+        protected override void Initialize(ODataContext context, ODataPath path)
+        {
+            base.Initialize(context, path);
+            _deleteRestriction = Restriction?.DeleteRestrictions ??
+                Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty, CapabilitiesConstants.DeleteRestrictions);
+        }
```

Comment:
I noticed these codes are repeated in all navigation property handler, could it be better in the base class?

Reply:
Yes. I have refactored these to be in the base class.